docs/strategy-advanced.md

```diff
@@ -0,0 +1,92 @@
+
+Simple, timebased order-timeouts can be configured either via strategy or in the configuration in the `unfilledtimeout` section.
+
+However, freqtrade also offers a custom callback for both ordertypes, which allows you to decide based on custom criteria if a order did time out or not.
+
+!!! Note
+    Unfilled order timeouts are not relevant during backtesting or hyperopt, and are only relevant during real (live) trading. Therefore these methods are only called in these circumstances.
+
+### Custom order timeout example
+
+A simple example, which applies different unfilled-timeouts depending on the price of the asset can be seen below.
+It applies a tight timeout for higher priced assets, while allowing more time to fill on cheap coins.
+
+The function must return either `True` (cancel order) or `False` (keep order alive).
+
+``` python
+from datetime import datetime, timestamp
+from freqtrade.persistence import Trade
+
+class Awesomestrategy(IStrategy):
+
+    # ... populate_* methods
+
+    # Set unfilledtimeout to 25 hours, since our maximum timeout from below is 24 hours.
+    unfilledtimeout = {
+        'buy': 60 * 25,
+        'sell': 60 * 25
+    }
+
+    def check_buy_timeout(self, pair: str, trade: 'Trade', order: dict, **kwargs) -> bool:
+        if trade.open_rate > 100 and trade.open_date < datetime.utcnow() - timedelta(minutes=5):
+            return True
+        elif trade.open_rate > 10 and trade.open_date < datetime.utcnow() - timedelta(minutes=3):
+            return True
+        elif trade.open_rate < 1 and trade.open_date < datetime.utcnow() - timedelta(hours=24):
+           return True
+        return False
+
+
+    def check_sell_timeout(self, pair: str, trade: 'Trade', order: dict, **kwargs) -> bool:
+        if trade.open_rate > 100 and trade.open_date < datetime.utcnow() - timedelta(minutes=5):
+            return True
+        elif trade.open_rate > 10 and trade.open_date < datetime.utcnow() - timedelta(minutes=3):
+            return True
+        elif trade.open_rate < 1 and trade.open_date < datetime.utcnow() - timedelta(hours=24):
+           return True
+        return False
+```
+
+!!! Note
+    For the above example, `unfilledtimeout` must be set to something bigger than 24h, otherwise that type of timeout will apply first.
+
+
+### Custom order timeout example (using additional data)
+
+``` python
+from datetime import datetime, timestamp
+from freqtrade.persistence import Trade
+
+class Awesomestrategy(IStrategy):
+
+    # ... populate_* methods
+
+    # Set unfilledtimeout to 25 hours, since our maximum timeout from below is 24 hours.
+    unfilledtimeout = {
+        'buy': 60 * 25,
+        'sell': 60 * 25
+    }
+
+    def check_buy_timeout(self, pair: str, trade: Trade, order: dict, **kwargs) -> bool:
+        ob = self.dp.orderbook(pair, 1)
+        current_price = ob['bids'][0][0]
+        # Cancel buy order if price is more than 2% above the order.
+        if order['price'] > current_price * 1.02:
```

Comment:
should be reversed? (to be "Cancel buy order if price is more than 2% above the order" as stated in the comment)
```
if current_price > order['price'] * 1.02:
```
???

(same for sell part)

Reply:
no ... it's correct this way (i got this wrong initially, too).

Assume i put my buy-order in at 100 USDT (using bids-side).
* price immediately jumps to 101 USDT (my trade is now in 15th spot in the orderbook ).
* Price jumps to 100 USDT - so my chances of beeing filled at 100 USDT are pretty slim -> i decide to cancel the order
therefore the calculation / comparsion is correct since it's: `102 > 100 * 1.02`.

If it where different - i place the buy-order at 100 - and price drops to 98 - then i'd have been filled already fully ... therefore that's no reason to cancel.

(the same applies to sell side - but obviously in the other direction.

Reply:
but 102 here is `current_price`, and `order['price']` is 100, so it should be
```
if current_price > order['price'] * 1.02:
```
to be 
```
102 > 100 * 1.02
```